javascript/collapsible.js

```diff
@@ -19,9 +19,88 @@ function collapseAllGroups() {
   }
 }
 
+var strclass = '';
+var res = '';
+$("tbody > tr").each(function() {
```

Comment:
What are you trying to achieve here?

Working with global variables like in this case, accessed by different handlers is a bad implementation. While I suggested using jQuery-handlers, that doesn't mean you need to remove functions entirely. Keep the code of the handlers in functions as much as possible and call those in the handlers. This is more readable than what you have now, functions can be documented easier etc.

So please describe what you are trying to achieve with this and the other handler and I'll try to find some better implementation without those global variables and twi functions being called by the handler.

Reply:
Regarding code style again: It doesn't make much sense to use code like the following:

	$("tbody > tr").each(function() {

		some stuff
	});

If you want the improved readability by the newline, simply use a style like the following:

	$("tbody > tr").each(function()
	{
		some stuff
	});

Your former style is more used by people who do NOT want the extra newline and save some visual space in the end. So the latter is more standard and widely known.

Reply:
There are two handlers 1 to run at the time of page load and 1 to run at the time on user operation.
1 - Page Load `$("tbody > tr").each(function() { });`
When the entire page is loaded with the data this handler will go through each of the table row. It will discard the rows that dont have `title` marked. The title attribute is used to generate the path and level information from the the file name that has been provided. From log file we get data as:
```
branches/ -> title = "branches"
branches/folder1/item1.txt - > title = "branches folder1"
branches/folder1.item2.txt - > title = "branches folder1"
branches/item3.txt - > title = "branches"
```
Aim of this handler to first identify unique folders at the root level and collapse all the items that are marked with that folder name. Here in this case branches. 
This will keep only root folders visible rest all will be collapsed.
This is how we get the first page with only root folders.

2 - User Operation - On Click Operation
This is used to expand the folders that are collapsed during the page load.
It is to attach to `td.path` as it is the dominant element visually. 
As on page load everything is closed the customaction is close. This is the attribute is used to track the operation performed on the folder.
When we click on say branches. 
It traverse through every row below it and keep making rows visible which all have the `branches` in the title. Meaning all files, and folders under branches. 
It exapands the subfolder in the branches as well. Giving an entire tree view. It marks all items with customaction as open as that is the action that has been performed.
Same logic is used to collapse or expand the subfolders udner the root folders. 

For the first handler as the handler is called each row there is a need to store value for a scope larger than the function call to track what is collapsing. Hence the variable
```
var strclass = '';
var res = ''; 
```
I am not able to figure it out how to do it some other way. Would be good know if a better way exists.

Reply:
While I didn't understand all of this, I have the feeling you are re-implementing CSS-selectors in the end and should instead simply use those in combination with registering click-events on individual dirs. There are selectors for e.g. searching individual words in HTML-attributes already, which seems to be exactly what you are doing with `title`:

	/* <a> elements whose class attribute contains the word "logo" */
	a[class~="logo"] {
	  padding: 2px;
	}

https://developer.mozilla.org/en-US/docs/Web/CSS/Attribute_selectors

State to distinguish what to do shouldn't be encoded in global variables unless absolutely necessary, which it doesn't seem to be in your case. When you know if to show/hide things, you can as well register click events on those elements to do something. This way you don't need state in variables, the pure fact that a click-handler is available doing what you want with this element is enough.

I'll try to have a closer look at the weekend.

Reply:
I looked at the link that you provided and after some search. I implemented an even simpler change. But that meant that i had to unified the `directory.tmpl` across all the three templates. Have a look at the new commit.